<?xml version="1.0"?>
<launch>
    <!-- Launching launch files: https://answers.ros.org/question/9426/can-i-call-one-launch-file-from-another-launch-file/-->
  <include file="$(find final_assignment)/launch/simulation_gmapping.launch" />
  <include file="$(find final_assignment)/launch/move_base.launch" />

<arg name="mod" default="mod2"/>

<group if="$(eval mod == 'mod1')">
  <arg name="des_pos_x" default= "-5.0"/>
  <arg name="des_pos_y" default= "-5.0"/>
  <param name="des_pos_x" value="$(arg des_pos_x)" />
  <param name="des_pos_y" value="$(arg des_pos_y)" />

  <node pkg="final_assignment" type="send_goal.py" name="send_goal" output="screen" launch-prefix="gnome-terminal --tab --title=&quot;Mod1: User Goal Input&quot; -- "/>

</group>

<group if="$(eval mod == 'mod2')">
  <!-- Teleopt- keyboard   https://answers.ros.org/question/231271/setting-argument-for-teleop-twist-keyboard-in-launch-file/-->
  <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop" output="screen" launch-prefix="gnome-terminal --tab --title=&quot;Mod2: Keyboard&quot; -- " />
</group>

<group if="$(eval mod == 'mod3')">
  <!-- Teleopt- keyboard   https://answers.ros.org/question/231271/setting-argument-for-teleop-twist-keyboard-in-launch-file/-->
  <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop" output="screen" launch-prefix="gnome-terminal --tab --title=&quot;Mod3: User Input with Coll. Avoid.&quot; -- " />
</group>


</launch>